src/prisma/cli/prisma.py

```diff
@@ -34,60 +28,60 @@ def run(
         'PRISMA_CLI_QUERY_ENGINE_TYPE': 'binary',
     }
     env = {**default_env, **env} if env is not None else default_env
-    # ensure the client uses our engine binaries
-    for engine in binaries.ENGINES:
-        env[engine.env] = str(engine.path.absolute())
 
-    if args and args[0] == 'studio':
-        click.echo(
-            click.style(
-                'ERROR: Prisma Studio does not work natively with Prisma Client Python',
-                fg='red',
-            ),
-        )
-        click.echo(
-            '\nThere are two other possible ways to use Prisma Studio:\n'
-        )
-        click.echo(
-            click.style('1. Download the Prisma Studio app\n', bold=True)
-        )
-        click.echo(
-            indent(
-                'Prisma Studio can be downloaded from: '
-                + click.style(
-                    'https://github.com/prisma/studio/releases', underline=True
-                ),
-                ' ' * 3,
-            )
-        )
-        click.echo(
-            click.style('\n2. Use the Node based Prisma CLI:\n', bold=True)
-        )
-        click.echo(
-            indent(
-                'If you have Node / NPX installed you can launch Prisma Studio by running the command: ',
-                ' ' * 3,
-            ),
-            nl=False,
-        )
-        click.echo(click.style('npx prisma studio', bold=True))
-        return 1
-
-    process = subprocess.run(
-        [str(path.absolute()), *args],
+    # TODO: ensure graceful termination
```

Comment:
One idea: You could maybe just make the ABC implementation responsible for the actual calling and have the implementation `run` methods just pass through their arguments and all the other foo from the signature. So that subprocess is called in just one place -- as a private method call on `Strategy`. That would force all exit code checking into one code path. Otherwise each strategy needs to concern itself not with building args but also with error checking

Reply:
Good idea - used this implementation strategy in 694dffbe343fd8930afd8423067248483398f71a.